File: LeetCode/Topicwise/Linked-List/005.reverse-linked-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var reverseList = function (head) {
  let cur = head;
  let fut = null;
  let prev = null;
  while (cur) {
    fut = cur.next;
    cur.next = prev;
    prev = cur;
    cur = fut;
  }
  head = prev;
  return head;
};

File: LeetCode/LeetCode - 🔝 Top Interview Questions/005.reverse-linked-list.py
class Solution:
    def reverseList(self, head: ListNode) -> ListNode:
        prevNode = None
        currNode = head
        nextNode = None
        
        while currNode:
            nextNode = currNode.next
            currNode.next = prevNode
            prevNode = currNode
            currNode = nextNode
        
        head = prevNode
        
        return head